### PR TITLE
Allow specification of job IDs in RemoteResults

### PR DESCRIPTION
**Pull request by HGSilveri**
- Allows specification of a subset of job IDs in `RemoteResults`
- Exposes `RemoteResults.batch_id` and `RemoteResults.job_ids`
- Supports specification of `job_ids` in `PasqalCloud._fetch_result()`

These changes should be useful to PRs #701 and #707 .

### CODE REVIEW

File: pulser-core/pulser/backend/remote.py [hunk 1]
```diff
@@ -58,18 +58,47 @@ class RemoteResults(Results):
...
+    ):
         """Instantiates a new collection of remote results."""
         self._submission_id = submission_id
         self._connection = connection
+        if job_ids is not None and not set(job_ids).issubset(
+            all_job_ids := self._connection._get_job_ids(self._submission_id)
+        ):
+            unknown_ids = [id_ for id_ in job_ids if id_ not in all_job_ids]
+            raise RuntimeError(
+                f"Submission {self._submission_id!r} does not contain jobs "
```
Comment (oliver-gordon) [inline pulser-core/pulser/backend/remote.py:80]: `!r` is there for inline quotes right? It's _not_ misplaced correct?

File: pulser-core/pulser/backend/remote.py [hunk 2]
```diff
@@ -102,7 +133,9 @@ def submit(
         pass
 
     @abstractmethod
-    def _fetch_result(self, submission_id: str) -> typing.Sequence[Result]:
+    def _fetch_result(
+        self, submission_id: str, job_ids: list[str] | None
+    ) -> typing.Sequence[Result]:
```
Comment (oliver-gordon) [inline pulser-core/pulser/backend/remote.py:138]: Not a related change but, how have you defined which types are imported via `typing.Type` or imported at the top? It's just the abstract classes?

File: pulser-pasqal/pulser_pasqal/pasqal_cloud.py [hunk 3]
```diff
@@ -189,7 +191,16 @@ def _fetch_result(self, submission_id: str) -> tuple[Result, ...]:
         meas_basis = seq_builder.get_measurement_basis()
 
         results = []
-        for job in batch.ordered_jobs:
+        sdk_jobs = batch.ordered_jobs
+        if job_ids is not None:
+            ind_job_pairs = [
```
Comment (oliver-gordon) [inline pulser-pasqal/pulser_pasqal/pasqal_cloud.py:196]: Just to confirm, this is so we can retain the returned order that they came in? if we associate and order by the index?

**Review by oliver-gordon — APPROVED** [review]
Some basic Q's but looks fine with me.

File: pulser-core/pulser/backend/remote.py (hunk 1 above)
Comment (HGSilveri) [reply pulser-core/pulser/backend/remote.py:80]: Yes, that's right

File: pulser-core/pulser/backend/remote.py (hunk 2 above)
Comment (HGSilveri) [reply pulser-core/pulser/backend/remote.py:138]: It's basically everything except `typing.Sequence` so as to not confuse it with `pulser.Sequence` (I know, it's unfortunate naming but there's no going back now)

File: pulser-pasqal/pulser_pasqal/pasqal_cloud.py (hunk 3 above)
Comment (HGSilveri) [reply pulser-pasqal/pulser_pasqal/pasqal_cloud.py:196]: Yes, whenever `job_ids` are provided we want to return the results in the same order they were given. This is simultaneously ensuring that's the case and selecting only the relevant subset of jobs

**Pull request merged**